=== modified file 'dwarf/dwarf_die_deliv.c'
--- dwarf/dwarf_die_deliv.c	2012-06-16 12:27:50 +0000
+++ dwarf/dwarf_die_deliv.c	2012-06-16 12:28:03 +0000
@@ -472,6 +472,7 @@
             dis->de_cu_context->cc_length +
             dis->de_cu_context->cc_length_size +
             dis->de_cu_context->cc_extension_size;
+		new_offset = (new_offset + 3) / 4 * 4;
     }
 
     /*  Check that there is room in .debug_info beyond the new offset
@@ -534,6 +535,7 @@
 
     new_offset = new_offset + cu_context->cc_length +
         cu_context->cc_length_size + cu_context->cc_extension_size;
+	new_offset = (new_offset + 3) / 4 * 4;
     *next_cu_offset = new_offset;
     return (DW_DLV_OK);
 }

